### ISSUE TITLE
[CP] Fix Tooltip crash when selected in a SelectableRegion

### ISSUE THREAD

**Issue by LongCatIsLooong** [flutter/flutter#131005]
### Issue Link

https://github.com/flutter/flutter/issues/129969

### Commit Hash

2da353a

### Target

beta

### PR Link

https://github.com/flutter/flutter/pull/130181

### Changelog Description

Application may crash when dragging mouse over Tooltip while selecting text in SelectionArea

### Impacted Users

Devs who uses Tooltip in a SelectionArea

### Impact Description

Application may crash when dragging mouse over Tooltip while selecting text in SelectionArea

### Workaround

Not always possible

### Risk

medium

### Test Coverage

yes

### Validation Steps

Follow the repro steps provided in https://github.com/flutter/flutter/issues/129969,
or run the following test:
```dart
  testWidgets('Tooltip is not selectable', (WidgetTester tester) async {
    const String tooltipText = 'AAAAAAAAAAAAAAAAAAAAAAA';
    String? selectedText;
    await tester.pumpWidget(
      MaterialApp(
        home: SelectionArea(
          onSelectionChanged: (SelectedContent? content) { selectedText = content?.plainText; },
          child: const Center(
            child: Column(
              children: <Widget>[
                Text('Select Me'),
                Tooltip(
                  message: tooltipText,
                  waitDuration: Duration(seconds: 1),
                  triggerMode: TooltipTriggerMode.longPress,
                  child: SizedBox.square(dimension: 50),
                ),
              ],
            ),
          ),
        ),
      ),
    );

    final TooltipState tooltipState = tester.state(find.byType(Tooltip));

    final Rect textRect = tester.getRect(find.text('Select Me'));
    final TestGesture gesture = await tester.startGesture(Alignment.centerLeft.alongSize(textRect.size) + textRect.topLeft);
    // Drag from centerLeft to centerRight to select the text.
    await tester.pump(const Duration(seconds: 1));
    await gesture.moveTo(Alignment.centerRight.alongSize(textRect.size) + textRect.topLeft);
    await tester.pump();

    tooltipState.ensureTooltipVisible();
    await tester.pump();
    // Make sure the tooltip becomes visible.
    expect(find.text(tooltipText), findsOneWidget);
    assert(selectedText != null);

    final Rect tooltipTextRect = tester.getRect(find.text(tooltipText));
    // Now drag from centerLeft to centerRight to select the tooltip text.
    await gesture.moveTo(Alignment.centerLeft.alongSize(tooltipTextRect.size) + tooltipTextRect.topLeft);
    await tester.pump();
    await gesture.moveTo(Alignment.centerRight.alongSize(tooltipTextRect.size) + tooltipTextRect.topLeft);
    await tester.pump();

    expect(selectedText, isNot(contains('A')));
  });
```

**Comment by itsjustkevin** [comment]
@LongCatIsLooong for the changelog description could you include the platforms this occurs on?  The issues mention that it is MacOS, so my assumption is that it crashes on other desktop platforms. 

Does this issue occur on the web?  

For guidance, please see: https://github.com/flutter/flutter/wiki/Hotfix-Documentation-Best-Practices

**Comment by itsjustkevin** [comment]
@HansMuller could you take a look at this cherry-pick request and attach the `cp: approved` label if you support this cherry-pick?

**Comment by LongCatIsLooong** [comment]
I believe this affects all platforms including the web since it's a framework issue.

**Comment by itsjustkevin** [comment]
@LongCatIsLooong where is the link the the cherry-pick PR here?

**Comment by LongCatIsLooong** [comment]
My apologies the PR is https://github.com/flutter/flutter/pull/130181. Updated issue.

**Comment by itsjustkevin** [comment]
@LongCatIsLooong this points to the original PR to master.

**Comment by LongCatIsLooong** [comment]
https://github.com/flutter/flutter/pull/131288

**Comment by CaseyHillers** [comment]
Shipped in 3.13.0-0.4.pre beta
